src/cargo/core/manifest.rs

```diff
@@ -209,6 +214,17 @@ pub struct Target {
     for_host: bool,
 }
 
+#[derive(Clone, PartialEq, Eq, Debug)]
+struct NonHashedPathBuf {
+    path: PathBuf,
+}
+
+impl Hash for NonHashedPathBuf {
+    fn hash<H: Hasher>(&self, _: &mut H) {
+        // ...
+    }
+}
```

Comment:
A slightly more elaborte design here would be to introduce a notion of explicitly relative path here and track path relative to workspace root and to other workspace members outside the root. Sort of like

```Rust
struct PathRoots {
  // a number of anchors, such as current workspace dir, a particular workspace member, users's home directory, 
  // CARGO_HOME, etc. This structure is "global" for cargo and is stored in Workspace/config.
  roots: Vec::<PathBuf>, 
}

struct RelativePath {
  root_idx: usize // index in the global PathRoots
  rel_path: PathBuf
}

impl RelativePath {
   fn to_path(&self, roots: &PathRoots) -> PathBuf {
    roots[self.root_idx].join(&self.rel_path)
  }
}
```

Not sure that we need this, but this can cope with workspaces which are not entirely under a workspace root. (Or perhaps it was a mistake to allow non-subdirectory members?).